lib/server.js

```diff
@@ -117,8 +119,29 @@ bedrock.events.on('bedrock-express.configure.routes', app => {
     routes.notify, brRest.when.prefers.ld,
     validate('continuity-server.notification'),
     asyncHandler(async (req, res) => {
-      // FIXME: we need to authenticate `peerId`; http-signatures should be
-      // added here
+      const {url, method, headers} = req;
+      const expectedHeaders = [
+        '(created)',
+        // (key-id) will include the peerId in the signature itself
+        '(key-id)',
+        '(request-target)',
+        'digest',
+        'content-type'
+      ];
+      const parsed = parseRequest(
+        {url, method, headers}, {headers: expectedHeaders});
+      // get parsed parameters from from HTTP header and generate signing string
+      const {keyId, signingString, params: {signature: b64Signature}} = parsed;
+      const key = await wrapKeyPair({peerId: keyId});
+      const verifier = key.verifier();
+      const encoder = new TextEncoder();
+      const data = encoder.encode(signingString);
+      const signature = Buffer.from(b64Signature, 'base64');
```

Comment:
Parsing the signature probably needs to be wrapped in a try/catch.

Reply:
I wrapped the entire function in a try/catch and made it a helper here: https://github.com/digitalbazaar/bedrock-ledger-consensus-continuity/pull/249/commits/2341a04e262986a6d37bfc9c52da3264f595a20a